common/persistence/sqlMetadataPersistence.go

```diff
@@ -0,0 +1,383 @@
+	_, err = m.GetDomain(&GetDomainRequest{Name: request.Info.Name})
+	switch err.(type) {
+	case *workflow.EntityNotExistsError:
+		// Domain does not already exist. Create it.
+
+		// Encode request.ReplicationConfig.Clusters
+		clusters, err := gobSerialize(serializeClusterConfigs(request.ReplicationConfig.Clusters))
+		if err != nil {
+			return nil, &workflow.InternalServiceError{
+				Message: fmt.Sprintf("CreateDomain operation failed. Failed to encode ReplicationConfig.Clusters. Value: %v", request.ReplicationConfig.Clusters),
+			}
+		}
+
+		if _, err := tx.NamedExec(templateCreateDomainSqlQuery, &FlatCreateDomainRequest{
+			DomainInfo:   *(request.Info),
+			DomainConfig: *(request.Config),
+			// TODO Extracting the fields from DomainReplicationConfig since we don't currently support
+			// TODO DomainReplicationConfig.Clusters
+
+			//DomainReplicationConfig: *(request.ReplicationConfig),
+			ActiveClusterName: request.ReplicationConfig.ActiveClusterName,
+			Clusters:          clusters,
+
+			// TODO
+
+			IsGlobalDomain:  request.IsGlobalDomain,
+			ConfigVersion:   request.ConfigVersion,
+			FailoverVersion: request.FailoverVersion,
+		}); err != nil {
+			print(err.Error())
+			return nil, &workflow.InternalServiceError{
+				Message: fmt.Sprintf("CreateDomain operation failed. Inserting into domains table. Error: %v", err),
+			}
+		}
+
+		tx.Commit()
+		return &CreateDomainResponse{ID: request.Info.ID}, nil
+
+	case nil:
+		// The domain already exists.
+		return nil, &workflow.DomainAlreadyExistsError{
+			Message: fmt.Sprintf("Domain already exists.  DomainId: %v", request.Info.ID),
+		}
+
+	default:
+		print("default")
+		print(err.Error())
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf(
+				"CreateDomain operation failed. Could not check if domain already existed. Error: %v", err),
+		}
+	}
+}
+
+func (m *sqlMetadataManager) GetDomain(request *GetDomainRequest) (*GetDomainResponse, error) {
+	// TODO cassandraMetaPersistence_test seems to lack a test case for when both ID and name are empty
+	// TODO (even though the current Cassandra implementation does handle it)
+	// TODO so for now we will not handle it either.
+
+	// CAVEAT! The Cassandra persistence implementation relies on the following behaviour of gocql's scan:
+	// if a column entry is NULL, the result of the scan is the type's default value
+	// e.g. if dbVersion is NULL, we will get dbVersion 0
+	// We have to explicitly implement this behaviour since the result of a scan
+	// with go-sql-driver/mysql is that
+	// the map[string]interface{} has, for its "dbVersion" key, an interface{}(nil) as the value
+
+	var rows *sqlx.Rows
+	var err error
+
+	if len(request.Name) > 0 {
+		if len(request.ID) > 0 {
+			return nil, &workflow.BadRequestError{
+				Message: "GetDomain operation failed.  Both ID and Name specified in request.",
+			}
+		} else {
+			rows, err = m.db.NamedQuery(templateGetDomainByNameSqlQuery, request)
```

Comment:
There is a db.Get method that does the binding for you.
BTW you should use db.Get or db.Select instead of named query directly for most cases.